core/charm/series_selector.go

```diff
@@ -59,6 +53,28 @@ type SeriesSelector struct {
 	// from bundle specifies the deploy request comes from a bundle spec.
 	FromBundle bool
 	Logger     SeriesSelectorLogger
+	// UsingImageID is true when the user is using the image-id constraint
+	// when deploying the charm. This is needed to validate that in that
+	// case the user is also explicitly providing a base.
+	UsingImageID bool
+}
+
+// TODO: The Force flag is really only valid if the SeriesFlag is specified. There is
+// code and tests that allow the Force flag when series isn't specified, but
+// they should really be cleaned up. The `deploy` CLI command has tests to
+// ensure that --Force is only valid with --series.
```

Comment:
Please update the TODO. Force in deploy can also be used to force an LXD profile which does not pass validation. LXD profile validation includes an allowed list, as we do not encourage all possible LXD profiles within charms.

Reply:
done